lib/core/request.js

```diff
@@ -51,13 +52,13 @@ class Request {
       method !== 'CONNECT'
     ) {
       throw new InvalidArgumentError('path must be an absolute URL or start with a slash')
-    } else if (invalidPathRegex.exec(path) !== null) {
+    } else if (invalidPathRegex.test(path)) {
       throw new InvalidArgumentError('invalid request path')
     }
 
     if (typeof method !== 'string') {
       throw new InvalidArgumentError('method must be a string')
-    } else if (!isValidHTTPToken(method)) {
+    } else if (normalizeMethodRecord[method] === undefined && !isValidHTTPToken(method)) {
```

Comment:
Is there a perf difference for doing 
```suggestion
    } else if (method in normalizeMethodRecord && !isValidHTTPToken(method)) {
```

Reply:
Unfortunately, `in operation` is slow to look up properties that do not exist.

Reply:
k, thanks